File: Python/_121_Best_Time_to_Buy_and_Sell_Stock/solution.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 1:
            return 0
        dp = []
        minimum = prices[0]
        maximum = 0
        dp.append(0)
        for i in range(1, len(prices)):
            if prices[i] < minimum:
                minimum = prices[i]
                dp.append(0)
            else:
                dp.append(prices[i] - minimum)
            if maximum < dp[i]:
                maximum = dp[i]
        return maximum
        

File: Java/_121_Best_Time_to_Buy_and_Sell_Stock/Solution.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length < 1)
            return 0;
        int minimum = prices[0];
        int maximum = 0;
        for (int i = 1; i < prices.length; i++)
        {
            int temp = 0;
            if (prices[i] < minimum) 
                minimum = prices[i];
            else
                temp = prices[i] - minimum;
            if (maximum < temp)
                maximum = temp;       
        }
        return maximum;
    }
}